test/test_utf8_sanitizer.rb

```diff
@@ -224,6 +224,21 @@ def sanitize_form_data(request_env = request_env)
       end
     end
 
+    it "sanitizes StringIO rack.input with form encoded bad encoding" do
+      input = "foo=bla&foo=baz&quux%ED=bar%ED"
+      @rack_input = StringIO.new input
+
+      sanitize_form_data do |sanitized_input|
+        sanitized_input.split("&").each do |pair|
```

Comment:
Worth mentioning `URI.decode_www_form` wasn't used here because it does its own encoding magic. `URI.decode_www_form_component` is just straight unescaping, though.